code/game/objects/items/devices/cable_painter.dm

```diff
@@ -0,0 +1,50 @@
+/*--      THOU HAV BEEN WARNED KNAVE       --*/
+/*--   FORE HERE THERE ARE MANY EVIL BUGS  --*/
+/*--      ENTER ON THINE OWNE ACCORDE      --*/
+/*-- Feel free to checkout our gift store! --*/ //>:^#)
+
+obj/item/device/cable_painter
+	name = "cable painter"
+	desc = "A device for repainting laid cables."
+	icon = 'icons/obj/bodybag.dmi'
+	icon_state = "bodybag_folded"
+	var/list/modes
+	var/mode
+	w_class = 2.0
+
+obj/item/device/cable_painter/New()
+	..()
+	modes[0] = COLOR_RED
+	modes[1] = COLOR_YELLOW
+	modes[2] = COLOR_GREEN
+	modes[3] = COLOR_BLUE
+	modes[4] = COLOR_PINK
+	modes[5] = COLOR_ORANGE
+	modes[6] = COLOR_CYAN
+	modes[7] = COLOR_WHITE
```

Comment:
I would set this list more like:
`modes["red"] = COLOR_RED`

That way later in attack_self() you can do 
`mode = input( "What color would you like to use?", "Choose a Color", null, null) in modes`